### DISCUSSION TITLE
Is there a way to react on WM_KILLFOCUS in flutter app

### DISCUSSION THREAD

**Discussion by sturkopf123** [halildurmus/win32#359] [Q&A]
In more detail, is it possible to react in flutter on "WM_KLIIFOCUS" such that for example the user of the app could get locked out again in order to prevent security issues.

**Comment by domesticmouse** [comment]
@sturkopf123 I suspect this issue has been posted in the wrong issue tracker. If you are asking questions about the Flutter Win32 embedder, the correct issue tracker to ask questions on is https://github.com/flutter/flutter/issues.

Alternatively, if you are asking about @timsneath's [`win32_runner`](https://github.com/timsneath/win32_runner), then the correct issue tracker is https://github.com/timsneath/win32_runner/issues.

**Reply by timsneath** [reply]
(Actually, we don't accept _questions_ at the Flutter issue tracker: it is only for reproducible _bugs_.)

Brett is right though -- in that the win32 package doesn't itself do anything of this nature. There are two approaches you might take:

- You can change the C++-based Win32 runner that hosts your Flutter code and add custom code to handle the `WM_KILLFOCUS` message. The Win32 message loop for a Flutter app is found in the `windows/runner/win32_window.cpp` file. 
- You could use a [Win32 hook](https://docs.microsoft.com/en-us/windows/win32/winmsg/about-hooks) to intercept this message using Dart and have it do something to the Flutter app. The APIs required are included in this package, although I don't think there's any prior art.

I would recommend the former approach for a professional-grade app, since hooks add overhead to every Win32 message.
